do_kafka_test.sh

```diff
@@ -0,0 +1,153 @@
+# !/bin/bash
+
+# This script is used to test the S3D application.
+# It starts Kafka, S3D, and pushes test data to Kafka.
+# It then kills S3D and Kafka.
+
+# Make sure to set the following environment variables:
+# - DOCKER_HOST_IP
+# - DEPOSIT_GROUP
+# - DEPOSIT_KEY_NAME
+# - DEPOSIT_BUCKET_NAME
+# - AWS_REGION
+# - DEPOSIT_TOPIC
+# - HEADER_X_API_KEY
+
+setup() {
+    # build if target folder does not exist
+    if [ ! -d "target" ]; then
+        echo "Building."
+        mvn clean package assembly:single
+    else
+        echo "Target folder exists. Skipping build."
```

Comment:
Just a thought here, but what if we made a change and want to test it? With this logic we'd have to manually rebuild before running the do_kafka_test script. Should we force a new build regardless?

Reply:
Also, we may want to run our checks before the build to avoid wait if something is wrong.

Reply:
That's true. It would be more correct to force a new build regardless. The reason I initially added this 'skip build' case was because the S3D seems to take a long time to compile. We should look at minimizing the dependencies of the project when we can.

I will modify this to force a build each time & move the checks.

Reply:
The `do_kafka_test.sh` script now builds the project each time the script is run and only after our preliminary environment variable checks are made.

Reply:
The `run_with_endpoint.sh` & `run.sh` scripts will only build the project if the target folder doesn't exist. Otherwise we'll be building the project twice when we run `do_kafka_test.sh`.